cookbooks/cdo-mysql/files/default/update_servers.sql

```diff
@@ -18,14 +25,14 @@ REPLACE INTO `mysql_servers` (hostgroup_id, hostname, port, status, weight)
 UPDATE `mysql_servers` SET `weight` = CASE
   WHEN ( -- If row is a writer (hostname in HG0)
     `hostname` IN (SELECT `hostname` FROM `mysql_servers` WHERE (`hostgroup_id` = 0))
-  ) THEN
-    -- Set writer weight low for HG1, high for HG2.
+  ) THEN -- Set writer weight low for HG1, high for HG2.
```

Comment:
@sureshc @Hamms I think this comment is a little misleading. What it actually is doing is "Set writer weight low for HG1, _otherwise_ high"

Is that what we want? For example, if the writer weight in HG0 is 1, or 12, or whatever, this code will increase it to 10000000. That isn't a problem necessarily, as it's the only instance in that group.

Options:
1. edit the comment to match the code
2. edit the code to match the comment via elseif (to avoid HG0 or HG3 changes)
3. change the logic to be "Set writer weight high for HG2, otherwise 1"

Option one changes little and avoids scope creep. Option 3 feels most appropriate and cleanest.

Reply:
It's not entirely clear to me what we *do* want here, but I agree that option 3 feels the best. What are the potential failure modes we need to be considering here?